newIDE/electron-app/app/main.js

```diff
@@ -18,6 +18,8 @@ const { loadModalWindow } = require('./ModalWindow');
 const { load, registerGdideProtocol } = require('./Utils/UrlLoader');
 const throttle = require('lodash.throttle');
 const { findLocalIp } = require('./Utils/LocalNetworkIpFinder');
+const discordClient = require('discord-rich-presence')('718194112775454720');
```

Comment:
Is this asking at some point for the user permission? How does it work for a Discord user? Is having the Discord app installed enough? (sorry I'm a total noob on this)

Reply:
It is seamless. The discord app has an IPC server running in the background. This connects to the discord app, "authenticates" the app using the app ID so that discord knows it is a registered discord app, and then establishes a long term communication with the app. As long as one doesn't get closed, the connection and therefore rich presence stays on.